modules/local_contacts/modules.php

```diff
@@ -68,6 +68,62 @@ public function process() {
             $this->session->record_unsaved('Contact Added');
             Hm_Msgs::add('Contact Added');
         }
+
+        list($success, $form) = $this->process_form(array('contact_source', 'add_contact'));
```

Comment:
Form needs to be processed only once and code above already does it. However, I think the functionality itself here is worth adding another handler class and method to process the form. Please separate it from the original handler method for adding a single contact.